site/content/posts/2023-08-08-v0.25.0-impersonation-proxy-with-external-certs.md

```diff
@@ -0,0 +1,318 @@
+tags: ['Joshua T. Casey','Ryan Richard', 'Benjamin Petersen', 'release']
+---
+
+![Friendly seal](https://images.unsplash.com/photo-1618075254460-429d47b887c7?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=2148&q=80)
+*Photo by [karlheinz_eckhardt Eckhardt](https://unsplash.com/@karlheinz_eckhardt) on [Unsplash](https://unsplash.com/s/photos/seal)*
+
+With Pinniped v0.25.0 you get the ability to configure an externally-generated certificate for the Impersonation Proxy to serve TLS.
+
+To read more on this feature, and the design decisions behind it, see the [proposal](https://github.com/vmware-tanzu/pinniped/tree/main/proposals/1547_impersonation-proxy-external-certs).
+To read more about the impersonation proxy, see the [docs](https://pinniped.dev/docs/reference/supported-clusters/#background).
+
+To see the feature in practice on a local kind cluster, follow these instructions.
+
+The setup: Using a kind cluster, Contour as an ingress to the Impersonation Proxy, and `cert-manager` to generate a TLS serving cert.
+
+```shell
+Docker desktop v1.20.1
+Kind v0.20.0
+Contour v1.25.2
+Pinniped v0.25.0
+pinniped CLI v0.25.0 (https://pinniped.dev/docs/howto/install-cli/)
+cert-manager v1.12.3
+````
+
+Set up kind to run with Contour, using the example kind cluster configuration file provided by Contour.
+
+```shell
+$ wget https://raw.githubusercontent.com/projectcontour/contour/main/examples/kind/kind-expose-port.yaml
+$ kind create cluster \
+    --config kind-expose-port.yaml \
+    --name kind-with-contour \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Install Contour (see https://projectcontour.io/getting-started/ for more details).
+
+```shell
+# From https://projectcontour.io/getting-started/
+$ kubectl apply \
+    --filename https://projectcontour.io/quickstart/contour.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+# Verify that the Contour pods are running
+$ kubectl get pods \
+    --namespace projectcontour \
+    --output wide \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Install Pinniped’s local-user-authenticator and add some sample users (see https://pinniped.dev/docs/tutorials/concierge-only-demo/ for more details).
+
+```shell
+# Install Pinniped's local-user-authenticator so that you have some demo users to play with
+$ kubectl apply \
+    --filename https://get.pinniped.dev/v0.25.0/install-local-user-authenticator.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+# Create a local user "pinny" with password "password123"
+$ kubectl create secret generic pinny \
+    --namespace local-user-authenticator \
+    --from-literal=groups=group-for-mtls \
+    --from-literal=passwordHash=$(htpasswd -nbBC 10 x password123 | sed -e "s/^x://") \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+# The local-user-authenticator will serve a TLS endpoint for the Concierge to talk to,
```

Comment:
Clarify why are we doing this? Maybe something like `you need the CA bundle of the local-user-authenticator to configure Pinniped's Concierge` or something.

Reply:
done, see next push